async :)

samirans89 opened the pull request:
- Successful test [build](https://automate.browserstack.com/dashboard/v2/builds/55ef4160a81fffee25d68ebe6e49405e4eedeb8b)
- Same tests already working on Android mobile.
- WDIO Types defined for custom commands
- Mochawesome report + Allure
- GitHub Actions

samirans89 committed:
async :)

abhi-singhs commented on __tests__/pages/checkoutPage.ts:
```diff
@@ -3,43 +3,37 @@ import { Page } from "./basePage";
-  /**
-   * define selectors using getter methods
-   */
-  get loginText(): WebdriverIO.Element {
```
Any particular reason for removing the getter method?

NightWing1998 commented on __tests__/suites/paypal-button-click-pom.test.ts:
```diff
@@ -2,26 +2,22 @@ import { expect } from "chai";
-describe("Testing Paypal buttons", () => {
-  beforeEach("Open StackDemo", () => {
-    browser.url("/demo/checkout/#/pattern/client");
+describe("Testing Paypal buttons", async () => {
```
no need for an async with describe. Can we confirm if this is hampering functionality in any way?

samirans89 replied:
done. it wont impact functionality. 
the wdio docs define it without async so we are good.
https://webdriver.io/docs/sync-vs-async/

NightWing1998 commented on resources/config/wdio.prod.browserstack.conf.ts:
```diff
@@ -3,78 +3,99 @@ import * as _ from "lodash";
 
 export const config = _.defaultsDeep(overrides, defaultConfig);
+
+config.capabilities.forEach(function (caps: { [x: string]: unknown }) {
+  for (const i in config.commonCapabilities)
+    caps[i] = caps[i] || config.commonCapabilities[i];
+});
```
can we refactor this? It looks weird to export something and then change its value

samirans89 replied:
updated.

samirans89 committed:
updates for review comments

samirans89 committed:
addressed review comments

samirans89 committed:
add getters

samirans89 committed:
final updates

NightWing1998 committed:
updated(package.json,package-lock.json): adding mochawesome report generators,
updated(resources/config/wdio.conf.ts): adding reporters and merging results using mergeResults from wdio-mochawesome-reporter

NightWing1998 committed:
created(.github/workflows): adding github workflow

NightWing1998 committed:
updated(.prettierignore): ignoring github folder

NightWing1998 committed:
updated(.github/workflow): moving 'on' attr of PR from branch to type,
updated(.prettierignore): not ignoring github workflows

NightWing1998 committed:
updating github actions

NightWing1998 committed:
updating github actions

NightWing1998 committed:
updating github actions

NightWing1998 committed:
updating github actions

NightWing1998 committed:
updated(resources/config/wdio.conf.ts): adding allure reporting,
updated(package.json): adding allure reporting

NightWing1998 committed:
Merge branch 'reporting' of github.com:NightWing1998/webdriver-io-boilerplate into github_actions

NightWing1998 committed:
updating OS in github actions

samirans89 committed:
Merge pull request #16 from NightWing1998/github_actions

Adding github actions
